### PR TITLE
LPS-172987 | Validation when importing content into Knowledge Base

### PR DESCRIPTION
**Pull request by jaimelr10**
Description: Apply adjustments to import exception styling and messages on knowledge base.

Jira Ticket: https://issues.liferay.com/browse/LPS-172987

In this Pull Request we also solve the bug reported in https://issues.liferay.com/browse/LPS-174319

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -16359,6 +16358,7 @@ the-connection-to-the-remote-live-site-cannot-be-established-due-to-a-network-pr
 the-content-is-not-valid=The content is not valid.
 the-content-references-a-missing-file-entry=The content references a missing file entry.
 the-content-references-a-missing-page=The content references a missing page.
+the-content-was-imported-but-no-articles-were-found-with-one-of-the-supported-extensions-x=The content was imported, but no articles were found with one of the supported extensions: {0}.
```
Comment (AliciaGarciaGarcia) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:16361]: It's a rule to make separate the commit https://github.com/liferay-lima/liferay-portal/pull/3954/commits/5c5ef6eb8675ee50822eb9625367678ace250063 on 2 separate commits: 

1: The change on the `Language.properties`
2: The build lang on a separate commit for the rest of the languages

**Review by AliciaGarciaGarcia — CHANGES_REQUESTED** [review]

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/import.jsp [hunk 2]
```diff
@@ -34,9 +34,11 @@ renderResponse.setTitle(LanguageUtil.get(resourceBundle, "import"));
...
 		<aui:input name="parentKBFolderId" type="hidden" value="<%= String.valueOf(parentKBFolderId) %>" />
 
-		<liferay-ui:error exception="<%= KBArticleImportException.MustHaveACategory.class %>">
-			<%= LanguageUtil.format(request, "an-unexpected-error-occurred-while-importing-articles-x", LanguageUtil.get(request, "there-are-one-or-more-mandatory-vocabularies-assigned-to-the-knowledge-base-article")) %>
-		</liferay-ui:error>
+		<div class="mt-4">
+			<liferay-ui:error exception="<%= KBArticleImportException.MustHaveACategory.class %>">
+				<%= LanguageUtil.format(request, "an-unexpected-error-occurred-while-importing-articles-x", LanguageUtil.get(request, "there-are-one-or-more-mandatory-vocabularies-assigned-to-the-knowledge-base-article")) %>
+			</liferay-ui:error>
+		</div>
```
Comment (adolfopa) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/import.jsp:41]: Just a question, not a change request: why are we styling this error message in a different way? Shouldn't all error message look the same?

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/import.jsp (hunk 2 above)
Comment (jaimelr10) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/import.jsp:41]: In this case the error message it's the same, the only thing that changes is it's possition on the page

**Comment by jaimelr10** [comment]
ci:test:sf

**Comment by jaimelr10** [comment]
ci:test:relevant

**Review by AliciaGarciaGarcia — APPROVED** [review]

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/import.jsp [hunk 3]
```diff
@@ -34,9 +34,11 @@ renderResponse.setTitle(LanguageUtil.get(resourceBundle, "import"));
 		<aui:input name="mvcPath" type="hidden" value="/admin/import.jsp" />
 		<aui:input name="parentKBFolderId" type="hidden" value="<%= String.valueOf(parentKBFolderId) %>" />
 
-		<liferay-ui:error exception="<%= KBArticleImportException.MustHaveACategory.class %>">
-			<%= LanguageUtil.format(request, "an-unexpected-error-occurred-while-importing-articles-x", LanguageUtil.get(request, "there-are-one-or-more-mandatory-vocabularies-assigned-to-the-knowledge-base-article")) %>
-		</liferay-ui:error>
+		<div class="mt-4">
```
Comment (ambrinchaudhary) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/import.jsp:37]: It is a bit strange to have this `div` envolving only this error. I tested it and looks necessary to have margin-top (however I would change it to `mt-3`, 16px is enough and looks better), so my suggestion is to move the rest of the message errors inside this div. In addition, for some reason it is not rendering correctly, look at the markup, the error is outside the `div`. Could you take a look please? 

Screenshot 2023-02-01 at 11 33 43

**Review by ambrinchaudhary — CHANGES_REQUESTED** [review]

**Comment by jaimelr10** [comment]
ci:test:sf

**Comment by jaimelr10** [comment]
ci:test:relevant

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/import.jsp (hunk 3 above)
Comment (jaimelr10) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/import.jsp:37]: I've moved all the errors inside the div and now the markup looks well

**Review by ambrinchaudhary — APPROVED** [review]
LGTM! Thanks @jaimelr10!

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/view.jsp [hunk 4]
```diff
@@ -112,13 +108,19 @@ KBArticleURLHelper kbArticleURLHelper = new KBArticleURLHelper(renderRequest, re
...
+						<clay:alert
+							displayType="warning"
+							message='<%= LanguageUtil.format(request, "the-content-was-imported-but-no-articles-were-found-with-one-of-the-supported-extensions-x", HtmlUtil.escape(StringUtil.merge(kbGroupServiceConfiguration.markdownImporterArticleExtensions(), StringPool.COMMA_AND_SPACE))) %>'
+							title="Alert"
+						/>
 					</c:otherwise>
 				</c:choose>
 			</c:if>
 
+		<liferay-site-navigation:breadcrumb
```
[the remark this answers is not in the document]
Comment (jaimelr10) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/view.jsp]: I solved it and didn't complain either :+1:

**Comment by ambrinchaudhary** [comment]
ci:forward

**Pull request closed without merging**